### PR TITLE
[ci] updated Elasticsearch versions in CI and docs

### PR DESCRIPTION
**Pull request by jameslamb**
This PR updates the versions of Elasticsearch tested against in CI, per the agreement in CONTRIBUTING.md. It also removes some unnecessary intermediate versions in `setup_local.sh`.

### CODE REVIEW

**Comment by jameslamb** [comment]
> Looks like you'll have to fix `.ci/seed_es_on_travis.sh` too!

oy yep, forgot that one! Maybe we'll do #168 one day

File: .ci/seed_es_on_travis.sh [hunk 1]
```diff
@@ -76,7 +76,21 @@ case "$ES_VERSION" in
...
+      ;;
+
+    "7.4.2")
+      export MAPPING_FILE=${ES7_MAPPING_FILE};
+      export ES_BINARY_URL="${ES5PLUS_ARCHIVE}/elasticsearch-$ES_VERSION-amd64.deb"
+      # overwrite SAMPLE_DATA_FILE to use the ES7-compliant data
+      export SAMPLE_DATA_FILE="${ES7_SAMPLE_DATA_FILE}"
+      ;;
+
+    "7.4.2")
```
Comment (austin3dickey) [inline .ci/seed_es_on_travis.sh]: Whoops:
```suggestion
    "7.5.2")
```

Comment (jameslamb) [reply .ci/seed_es_on_travis.sh]: glad to have you reviewing my code again   @austin3dickey 😀

**Comment by jameslamb** [comment]
oh cool it passed! I thought "oh no, 7.5.2 is gonna have some weird thing in it that we have to deal with". Very happy that it didn't!

**Pull request merged**